daemon/emer-daemon.c

```diff
@@ -211,15 +219,85 @@ is_uuid (GVariant *variant)
 }
 
 static void
-uuid_from_variant (GVariant *variant,
-                   uuid_t    uuid)
+buffer_event (EmerDaemon *self,
+              GVariant   *event)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  g_variant_ref_sink (event);
+  gsize size_with_type = get_size_with_type (event);
+  gsize new_bytes_buffered = priv->num_bytes_buffered + size_with_type;
+  if (new_bytes_buffered > priv->max_bytes_buffered ||
+      /* Don't get wedged by a large event. */
+      new_bytes_buffered > MAX_REQUEST_PAYLOAD)
```

Comment:
Does this check not always restrict the total size regardless of what is set in `priv->max_bytes_buffered`? Maybe you intended to check `size_with_type > MAX_REQUEST_PAYLOAD`?